src/view-backend-private.cpp

```diff
@@ -139,6 +139,9 @@ void ViewBackend::unregisterSurface(uint32_t surfaceId)
 
     clearFrameCallbacks();
 
+    if (m_client.object)
+        wl_client_destroy(m_client.object);
```

Comment:
I would rather use `g_clear_pointer()`, which will do the `NULL` check, and also set the pointer to `NULL` after calling `wl_client_destroy()`:

```suggestion
    g_clear_pointer(&m_client.object, wl_client_destroy);
```

Reply:
Requested change incorporated to the patch. Thanks